### DISCUSSION TITLE
Upgrade 4.1.1 > 5.5.6  - unable to produce config file - it also complains about checking the db.

### DISCUSSION THREAD

**Discussion by AndKe** [ampache/ampache#3440] [Q&A]
I rm'ed my /var/www/ampache (4.1.1)
installed new 5.5.6 to same directory.(php 8.1)

I skip creating the database step. and proceed to the "generate config file" step:

![Screenshot from 2023-01-05 19-46-23](https://user-images.githubusercontent.com/1468842/211036509-c14b7677-ce8e-4842-bc6a-c5e80f0c1f25.png)

then I get this:
![Screenshot from 2023-01-05 19-48-26](https://user-images.githubusercontent.com/1468842/211036657-1e2bbb8e-70f3-4f93-a864-cd982a2ceafb.png)

Downloading htaccess works fine, but the download button for ampache.cfg.php just produces a blank webpage(no download)
clicking "write" does not work - (even if the config dir is 777)
..and some of those actions produce a warning line that the SQL test failed. - which I do not trust - as I've verified the database name, user name, and reset password.

**Comment by Andrew-McGee** [comment]
Not sure if you got past this but will reply for anyone else who may read this. A couple of things to check off the top of my head: 

- SELinux if you are on Fedora or RHEL - could be blocking access to files
- Check if you have the default ampache.cfg.php.dist - you can make a copy and edit this manually
- Try changing the owner of the directory and files to the same user apache runs as (www-data on Ubuntu; apache on Fedora)

**Comment by AndKe** [comment]
@Andrew-McGee Thank you - I don't even remember exactly how I solved it - but I think I dff'ed the default config file vs my backup from previous version.

**Reply by lachlan-00** [reply]
i'll see if i can recreate.

when you upgrade you should keep your config file and the database. The update mechanisms will maintain both those files.

But it's not a very common process to do so i;ll see if i can make it do the same thing.

**Reply by lachlan-00** [reply]
I think this may still be a permission issue as i just tested the following after backing up a config file

* deleted the config file
* started the install
* skipped create database
* clicked create config
 
![image](https://user-images.githubusercontent.com/1305249/221758196-20b7f9ea-3068-44d2-84af-0f889c22a7c7.png)

Then i was sent straight to the login screen. not sure what it could be other than permissions
